app/src/main/java/com/example/sharewhatyoucanproject/login/LoginFragment.kt

```diff
@@ -0,0 +1,96 @@
+package com.example.sharewhatyoucanproject.login
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.provider.Settings
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.example.sharewhatyoucanproject.databinding.FragmentLoginBinding
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.showToast
+
+class LoginFragment : Fragment() {
+
+    private var _binding: FragmentLoginBinding? = null
+    private val binding get() = _binding!!
+
+    private lateinit var loginViewModel: LoginViewModel
+
+    private val args: LoginFragmentArgs by navArgs()
+
+
+
+    override fun onCreate(savedInstanceState: Bundle?)
+    {
+
+        super.onCreate(savedInstanceState)
+
+        loginViewModel = ViewModelProvider(
+            this,
+            LoginViewModelFactory(activity?.application),
+        )[LoginViewModel::class.java]
+     }
+
+    @SuppressLint("HardwareIds")
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View? {
+
+        _binding = FragmentLoginBinding.inflate(inflater, container, false)
+
+        (activity as AppCompatActivity?)!!.supportActionBar!!.hide()
+
+        loginViewModel.type = UserType.values()[args.type]
+
+        loginViewModel.authenticationResult.observe(requireActivity()) { result ->
```

Comment:
don't observe using `requireActivity`, that means you are passing the activity lifecycle instead of the fragment's lifecycle and you will get a memory leak. Use `viewLifecycleOwner` like you have in other fragments

Reply:
yeah got it. Changed to viewLifecycleOwner.